src/Log/LogService.php

```diff
@@ -0,0 +1,209 @@
+        } else {
+            $this->logFilePath = $logPath . $logFileName;
+        }
+        return $this->logFilePath;
+    }
+
+    public function getDefaultLog()
+    {
+        if (array_key_exists('default.log', $this->loggers)) {
+            return $logger = $this->loggers['default.log'];
+        } else {
+            return $this->setDefaultLog('default.log');
+        }
+    }
+
+    public function setDefaultLog($logFileName)
+    {
+        $this->openLog('Zend_Config', $logFileName);
+        return $this->addLog();
+    }
+
+    public function addLog($logger = null)		//Done change to addLog (shouldn't be necessary)
+    {
+        if ($logger == null) {
+            $logger = $this->createLog();
+        }
+
+        if ($this->checkLogLevel() == false) {
+            $logLevelName = 'INFO';
+            $logger->warn('Log level not configured, using \'' . $logLevelName . '\'.');
+        } else {
+            $logLevelName = $this->checkLogLevel();
+        }
+
+        $zendLogRefl = new \ReflectionClass('Zend_Log');
+
+        $invalidLogLevel = false;
+
+        $logLevel = $zendLogRefl->getConstant($logLevelName);
+
+        if (empty($logLevel)) {
+            $logLevel = Zend_Log::INFO;
+            $invalidLogLevel = true;
+        }
+
+        // filter log output
+        $priorityFilter = new \Zend_Log_Filter_Priority($logLevel);
+        \Zend_Registry::set('LOG_LEVEL', $logLevel);
+        $logger->addFilter($priorityFilter);
+
+        if ($invalidLogLevel) {
+            $logger->err('Invalid log level \'' . $logLevelName .
+                '\' configured.');
+        }
+
+        $this->loggers[$this->logFileName] = $logger;      //how to know the logName for key value loggers other than logFileName
+        return $logger;
+    }
+
+    public function getLog($logName)		
+    {
+        if (array_key_exists($logName, $this->loggers)) {
+            return $logger = $this->loggers[$logName];
+        } else {
+            try {
+                $this->openLog('Zend_Config', $logName);
+                return $this->addLog();
+            } catch {
+                throw new Exception("Cannot create new log", 1);
+
+            }
+        }
+    }
+
+    public function checkLogLevel()
+    {
+        return isset($this->config->logging->log->level) ? $this->config->logging->log->level : false;
+    }
+
+    protected function createLog()
+    {
+        if ($this->format == null) {
+            $this->setDefaultFormat();
+        }
+
+        $formatter = new \Zend_Log_Formatter_Simple($this->format);
+
+        $writer = new \Zend_Log_Writer_Stream($this->logFile);
+        $writer->setFormatter($formatter);
+
+        $logger = new \Zend_Log($writer);
+        return $logger;
+    }
+
+    public function setDefaultFormat($format = null)		//DONE rename to setDefaultFormat()
+    {
+        $GLOBALS['id_string'] = uniqid(); // Write ID string to global variables, so we can identify/match individual runs.
```

Comment:
Would it make sense to move this code to `getDefaultFormat`, so you don't have to `setDefaultFormat` in order to get the __default__ format? Basically return the default format if nothing else has been set? That works similar to the getInstance function, but without any static stuff.

Reply:
Yes, it makes sense. We can keep a `setFormat()` function for custom format. Then we can check if there's a custom format. If not, we can call `getDefaultFormat()`.